src/libraries/System.Net.Primitives/src/System/Net/IPAddress.cs

```diff
@@ -74,8 +74,13 @@ internal uint PrivateAddress
         {
             get
             {
-                Debug.Assert(IsIPv4);
-                return _addressOrScopeId;
+                Debug.Assert(IsIPv4 || IsIPv4MappedToIPv6);
+                if (IsIPv4)
+                {
+                    return _addressOrScopeId;
+                }
+                uint address = (uint)_numbers[6] << 16 | (uint)_numbers[7];
+                return (uint)HostToNetworkOrder(unchecked((int)address));
```

Comment:
Nit: This property is also used by several `IPAddress` methods and it's somewhat untrivial that their behavior is unimpacted by this branch + the extra code in this method is unnecessary for those methods since they use the property only if `IsIPv4` is true. Would it make sense to keep this property simple and define an `IPNetwork.IPv4Address` property with the proposed behavior?

Reply:
I don't know if there is impact. But it is easy enough to split it so I did.